jack/pcm_jack.c

```diff
@@ -149,13 +164,16 @@ snd_pcm_jack_process_cb(jack_nframes_t nframes, snd_pcm_ioplug_t *io)
 	}
 
 	if (io->state == SND_PCM_STATE_RUNNING) {
-		areas = snd_pcm_ioplug_mmap_areas(io);
+		const snd_pcm_channel_area_t *areas = snd_pcm_ioplug_mmap_areas(io);
 
 		while (xfer < nframes) {
 			snd_pcm_uframes_t frames = nframes - xfer;
-			snd_pcm_uframes_t offset = jack->hw_ptr;
+			snd_pcm_uframes_t offset = jack->hw_ptr % io->buffer_size;
 			snd_pcm_uframes_t cont = io->buffer_size - offset;
 
+			/* split the snd_pcm_area_copy() function into two parts
+			 * if the data to copy passes the buffer wrap around
+			 */
```

Comment:
nice to add comment, but it does not belong to the commit topic...

Reply:
Moved to https://github.com/twischer-adit/alsa-plugins/pull/1/commits/30b9a26f0a186e9565aa38567e4cfe26c9645a57